### PR TITLE
server: implemented json.arrappend command

### PR DESCRIPTION
**Pull request by b0bleet**
Signed-off-by: b0bleet@placeq.com

### CODE REVIEW

**Comment by romange** [comment]
@iko1  if it's ok with you, I am asking you to review.

**Comment by romange** [comment]
@b0bleet,  do you need this command because you use Dragonfly, or you just wanted to contribute?
Can you please add yourself to CONTRIBUTORS.md ?

**Comment by b0bleet** [comment]
> @b0bleet, do you need this command because you use Dragonfly, or you just wanted to contribute? Can you please add yourself to CONTRIBUTORS.md ?

Hi @romange, Thanks for your reply. Actually, I want to contribute to DragonFly because I'm very interested in in-memory database development. At the moment I do review DragonFly's internal structure to contribute core components.

**Comment by romange** [comment]
Then join our discord and ping me there

File: src/server/json_family.cc [hunk 1]
```diff
@@ -778,6 +778,39 @@ OpResult<vector<OptSizeT>> OpArrInsert(const OpArgs& op_args, string_view key, s
   return vec;
 }
 
+// Returns numeric vector that represents the new length of the array at each path, or Null reply 
+// if the matching JSON value is not an array.
+OpResult<vector<OptSizeT>> OpArrAppend(const OpArgs& op_args, string_view key, string_view path,
+                                       const vector<json> append_values) {
```
Comment (iko1) [inline src/server/json_family.cc:784]: pass the vector by const-ref

Comment (b0bleet) [reply src/server/json_family.cc]: done

File: src/server/json_family.cc [hunk 2]
```diff
@@ -778,6 +778,39 @@ OpResult<vector<OptSizeT>> OpArrInsert(const OpArgs& op_args, string_view key, s
   return vec;
 }
 
+// Returns numeric vector that represents the new length of the array at each path, or Null reply 
+// if the matching JSON value is not an array.
```
Comment (iko1) [inline src/server/json_family.cc:782]: This line is redundant. I'd write only the return value type after a successful operation.

**Comment by iko1** [comment]
@b0bleet, Please make sure you squash the commits into a single commit and run `clang-format -i` on the changed files in the codebase.

**Comment by b0bleet** [comment]
> @b0bleet, Please make sure you squash the commits into a single commit and run `clang-format -i` on the changed files in the codebase.

done
thank you

**Review by iko1 — APPROVED** [review]
LGTM

File: src/server/json_family.cc [hunk 3]
```diff
@@ -778,6 +778,39 @@ OpResult<vector<OptSizeT>> OpArrInsert(const OpArgs& op_args, string_view key, s
...
+      vec.emplace_back(nullopt);
+      return;
+    }
+    for (auto& new_val : append_values) {
+      val.emplace_back(new_val);
+    }
+    vec.emplace_back(val.size());
+  };
+
+  json j = result.value();
```
Comment (dranikpg) [inline src/server/json_family.cc:803]: Just a side note: Doesn't this use the copy constructor and not the move one? (in all occurrences)

Comment (romange) [reply src/server/json_family.cc:803]: Yes, seems you found a performance bug.

Comment (romange) [reply src/server/json_family.cc:803]: Yes, it should be fixed.

File: src/server/json_family.cc [hunk 4]
```diff
@@ -763,7 +763,7 @@ OpResult<vector<OptSizeT>> OpArrInsert(const OpArgs& op_args, string_view key, s
     vec.emplace_back(val.size());
   };
 
-  json j = result.value();
+  json j = std::move(result.value());
```
Comment (iko1) [inline src/server/json_family.cc:766]: leading namespace here std::' is redundant and also for the other lines.

**Review by dranikpg — APPROVED** [review]

**Pull request merged**